### ISSUE TITLE
firmware filter for devices should populate from devices

### ISSUE THREAD

**Issue by jjcarstens** [nerves-hub/nerves_hub_web#684]
The `firmware` filter of the device index page is not that useful. It only lists firmware versions of which you still have a firmware record for. But firmwares get cleaned up and sometimes you have devices connecting which are on an old firmware that doesn't exist.

This makes it so you can't filter devices on reported firmware versions. You also can't filter devices that are "unknown" firmware (have yet to connect). In the screenshot below, you can see I have multiple firmwares that aren't listed in the firmware filter dropdown

<img width="492" alt="image" src="https://user-images.githubusercontent.com/11321326/98026735-d2122000-1dc8-11eb-9066-5e00c8adb855.png">

It would be nice if we could populate the filter option based on all versions reported from devices on the product.

**Comment by Mrjaco12** [comment]
I agree that updating this in some way would be a good idea. I just want to add context here about how we arrived at the way it works now. Since the live view only tracks devices that are being displayed, generating a list of firmware ids from the list of devices that the liveview knows about is not sufficient. The alternative is to fetch firmware versions for each device when the liveview mounts, which we decided was kinda expensive and it wouldn't necessarily have all the correct data if devices were added or removed while the user was on the page. It was also slightly more complicated to factor in filtering for the "unknown" state. So given all that we elected to use the easiest approach.

To summarize, there are a few considerations about tradeoffs that need to be decided before this feature can be fully implemented but I think this issue is bringing things in the right direction.

**Comment by jjcarstens** [comment]
Personally, I think its a worthwhile tradeoff to do a single DB query to select a single field from applicable device firmwares for the versions on mount. Or, it can be done when opening the "show filter" button as well.
